src/esp/assets/ResourceManager.cpp

```diff
@@ -2184,5 +2185,209 @@ std::unique_ptr<MeshData> ResourceManager::createJoinedCollisionMesh(
+        loadObjectMeshDataFromFile(filename, filename, "render", true);
+
+  }  // if no render asset exists
+
+  // fill points and triangles with mesh data
+  std::vector<float> points = std::vector<float>();
+  std::vector<uint32_t> triangles = std::vector<uint32_t>();
+  getPrimitiveMeshData(filename, points, triangles);
+
+  // use VHACD
+  bool res =
+      interfaceVHACD->Compute(&points[0], (unsigned int)(points.size() / 3),
+                              (const uint32_t*)&triangles[0],
+                              (unsigned int)(triangles.size() / 3), params);
+
+  Cr::Utility::Debug() << "== VHACD ran ==";
+
+  // convert convex hulls into MeshDatas, CollisionMeshDatas
+  int meshStart = meshes_.size();
+  std::vector<CollisionMeshData> collisionMeshGroup;
+  int nConvexHulls = interfaceVHACD->GetNConvexHulls();
+  Cr::Utility::Debug() << "Num Convex Hulls: " << nConvexHulls;
+  Cr::Utility::Debug() << "Resolution: " << params.m_resolution;
+  VHACD::IVHACD::ConvexHull ch;
+  std::unique_ptr<GenericMeshData> genCHMeshData;
+  for (unsigned int p = 0; p < nConvexHulls; ++p) {
+    // for each convex hull, transfer the data to a newly created  MeshData
+    interfaceVHACD->GetConvexHull(p, ch);
+
+    std::vector<Magnum::UnsignedInt> indices;
+    std::vector<Magnum::Vector3> positions;
+
+    // add the vertices
+    positions.resize(ch.m_nPoints);
+    for (size_t vix = 0; vix < ch.m_nPoints; vix++) {
+      positions[vix] =
+          Magnum::Vector3(ch.m_points[vix * 3], ch.m_points[vix * 3 + 1],
+                          ch.m_points[vix * 3 + 2]);
+    }
+
+    // add indices
+    indices.resize(ch.m_nTriangles * 3);
+    for (size_t ix = 0; ix < ch.m_nTriangles * 3; ix++) {
+      indices[ix] = ch.m_triangles[ix];
+    }
+    Cr::Containers::Optional<Mn::Trade::MeshData> CHMesh = Mn::MeshTools::owned(
+        Mn::Trade::MeshData{Mn::MeshPrimitive::Triangles,
+                            {},
+                            std::move(indices),
+                            Mn::Trade::MeshIndexData{indices},
+                            {},
+                            std::move(positions),
+                            {Mn::Trade::MeshAttributeData{
+                                Mn::Trade::MeshAttribute::Position,
+                                Cr::Containers::arrayView(positions)}}});
+    // Create a GenericMeshData
+    genCHMeshData = std::make_unique<GenericMeshData>(false);
+    genCHMeshData->setMeshData(*std::move(CHMesh));
+    genCHMeshData->BB = computeMeshBB(genCHMeshData.get());
+    genCHMeshData->uploadBuffersToGPU(false);
+
+    // Create CollisionMeshData and add to collisionMeshGroup vector
+    CollisionMeshData CHCollisionMesh = genCHMeshData->getCollisionMeshData();
+    collisionMeshGroup.push_back(CHCollisionMesh);
+
+    // register GenericMeshData in meshes_ dict
+    meshes_.emplace(meshes_.size(), std::move(genCHMeshData));
+  }
+  // make MeshMetaData
+  int meshEnd = meshes_.size() - 1;
```

Comment:
Doesn't this introduce narrowing conversion errors? Should be unsigned?

Reply:
I pass meshStart and meshEnd into the MeshMetaData which only accepts ints.